### ISSUE TITLE
Wiki Pages

### ISSUE THREAD

**Issue by Prakash2403** [iiitv/algos#182]
As ML and AI algorithms are being included in this repository, I think it is better to have our own Wiki for this repository. Also, as @aashutoshrathi has suggested, we could add application of algorithms there. What are your views @aviaryan @singhpratyush @Monal5031 @yashLadha ?

**Comment by singhpratyush** [comment]
Since the algorithms implemented are not industry standard, they will not be used by anyone. They are present here just to test the proof of concept. I think that just the comments in the implementation will do the job.

**Comment by Monal5031** [comment]
@Prakash2403 yes it seems a good idea to have that but in a little different way according to me.
We can create our own questions and add it there along with the dependent algorithm.
We can also use it for uploading questions for any future contests in our college and uploading answers there only along with used algo/DS.

**Comment by Prakash2403** [comment]
@singhpratyush There are multiple things which are too big to be explained in comments. For example, Average case time complexity of quick sort is `O(nlogn).` Proof using probability requires a wiki page. Similarly, when A* search algorithm will be added, then heuristic function and conditions required (admissibility and consistency) will require a wiki page. 
Although, these things can found on Internet, but it will be nice if we just give a brief summary here.

@Monal5031 Nice idea. But I think we should make a different repository dedicated to it.

What are your views @singhpratyush @aviaryan @yashLadha @aashutoshrathi @harishkrupo ?

**Comment by singhpratyush** [comment]
@Prakash2403: That is true. But this repository is to show the implementations. Things like consistency check, proof of complexity, etc. can be referenced from other destinations on the Internet.

But if the contributor really wants to give his words on the implementation, he may add a `README.md` in `docs/` directory of the repository.

@aashutoshrathi @aviaryan @Monal5031: What do you guys think of this?

**Comment by Monal5031** [comment]
@singhpratyush Yes sir it seems a nice idea, with the new contributors joining in it'd be good if that person explains their implementation if they did in a different manner. With the explanation being available on the algo/DS this repo would be more useful for people who are trying to learn new concepts.
Also sir we can use github pages to publish the explanations, like the main page could list all the algo/DS which has the explanations and others with the actual explanation. Views? @aviaryan @Prakash2403 @yashLadha @aashutoshrathi @mohitkyadav

**Comment by Prakash2403** [comment]
I agree. But if we are going to do this, we must do it asap, as this repository is growing bigger day by day.
